server/src/main/java/org/elasticsearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -109,6 +129,11 @@ public void accept(int value) {
                     + "] but was [" + count + "]. This limit can be set by changing the [" +
                     MAX_BUCKET_SETTING.getKey() + "] cluster level setting.", limit);
             }
+
+            if (value > 0 && checkBucketsStepSizeLimit > 0 && count % checkBucketsStepSizeLimit == 0) {
+                CircuitBreaker breaker = circuitBreakerService.getBreaker(CircuitBreaker.REQUEST);
```

Comment:
Instead of only getting the request circuit breaker here, can you take a CircuitBreaker instead of a CircuitBreakerService in the constructor?